prow/git/v2/client_factory.go

```diff
@@ -308,6 +315,33 @@ func (c *clientFactory) ClientFor(org, repo string) (RepoClient, error) {
 	return repoClient, nil
 }
 
+// Ensures that the repos in the cache are valid, clean, and up to date
+func (c *clientFactory) ensureValidUpdatedCache(cacheDir string, repoClient RepoClient, cacheClientCacher cacher) error {
+	// We only need to verify that the repos are valid once on startup
+	if _, ok := c.verifiedRepos[cacheDir]; !ok {
+		// Ensure it is valid
+		if valid, _ := repoClient.Fsck(); !valid {
```

Comment:
fsck needs to return an error because other implementations of the gitClientFactory need to return an error (namely the adapter needs to return a `notImplemented` error) For the implementation nil error will always also return true and non nil error will always return false.